src/coord/src/coord.rs

```diff
@@ -296,54 +296,13 @@ where
                     .with_context(|| format!("recreating sink {}", name))?;
                     coord.handle_sink_connector_ready(*id, connector);
                 }
-                CatalogItem::Index(index) => match id {
-                    GlobalId::User(_) => {
-                        // If index is on a local source, don't create a dataflow.
-                        if let CatalogItem::Source(Source {
-                            connector: SourceConnector::Local,
-                            desc,
-                            ..
-                        }) = coord.catalog.get_by_id(&index.on).item()
-                        {
-                            coord.views.insert(*id, ViewState::new(false, vec![]));
-                            broadcast(
-                                &mut coord.broadcast_tx,
-                                SequencedCommand::CreateLocalInput {
-                                    name: name.to_string(),
-                                    index_id: *id,
-                                    index: IndexDesc {
-                                        on_id: *id,
-                                        keys: index.keys.clone(),
-                                    },
-                                    on_type: desc.typ().clone(),
-                                },
-                            );
-                            coord.insert_index(*id, &index, Some(1_000));
-                        } else {
-                            coord.create_index_dataflow(name.to_string(), *id, index.clone())
-                        }
-                    }
-                    GlobalId::System(_) => {
-                        // TODO(justin): do something better here?
-                        if BUILTINS.tables().any(|t| t.index_id == *id) {
-                            let entry = coord.catalog.get_by_id(&index.on);
-                            coord.views.insert(*id, ViewState::new(false, vec![]));
-                            broadcast(
-                                &mut coord.broadcast_tx,
-                                SequencedCommand::CreateLocalInput {
-                                    name: name.to_string(),
-                                    index_id: *id,
-                                    index: IndexDesc {
-                                        on_id: *id,
-                                        keys: index.keys.clone(),
-                                    },
-                                    on_type: entry.desc().unwrap().typ().clone(),
-                                },
-                            );
-                        }
-                        coord.insert_index(*id, &index, Some(1_000))
+                CatalogItem::Index(index) => {
+                    if BUILTINS.logs().any(|log| log.index_id == *id) {
+                        coord.insert_index(*id, &index, Some(1_000));
```

Comment:
why are log indexes special cased?

Reply:
Good question! They are indexes that are created by the `EnableLogging` command, so they already exist on the dataflow workers. So we don't want to go through the usual path, which teaches the coordinator about the indexes _and_ creates those indexes on the workers—instead we just want to install them into the coordinator because they already exist on the workers.

I'll add a comment to this effect in a follow-up PR.